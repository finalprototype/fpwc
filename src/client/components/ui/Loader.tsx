import React from 'react';
import classNames from 'classnames';

import styles from './styles/Loader.scss';

interface Props {
  color?: string;
  inline?: boolean;
  size?: number;
  className?: string;
}

const Loader: React.FunctionComponent<Props> = (props: Props) => {
  const {
    className,
    inline,
    size,
    color,
  } = props;

  const containerClasses = classNames(
    styles.container,
    { [styles.inline]: inline },
    className,
  );

  return (
    <div
      className={containerClasses}
      style={{ width: `${size}px`, height: `${size}px` }}
    >
      <div
        style={{
          borderColor: `${color}`,
          borderWidth: size * 0.05,
        }}
      />
      <div
        style={{
          borderColor: `${color}`,
          borderWidth: size * 0.05,
        }}
      />
    </div>
  );
};

Loader.defaultProps = {
  color: '#ffffff',
  inline: false,
  size: 64,
  className: undefined,
};

export default Loader;
