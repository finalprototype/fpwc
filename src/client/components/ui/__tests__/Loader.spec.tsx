import React from "react";
import { shallow } from "enzyme";

import Loader from "../Loader";

describe("Loader", () => {
  it("renders default props", () => {
    const wrapper = shallow(<Loader />);
    expect(wrapper).toMatchSnapshot();
  });
});
