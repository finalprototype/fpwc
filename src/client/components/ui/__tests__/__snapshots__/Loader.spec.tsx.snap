// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader renders default props 1`] = `
<div
  className="container"
  style={
    Object {
      "height": "64px",
      "width": "64px",
    }
  }
>
  <div
    style={
      Object {
        "borderColor": "#ffffff",
        "borderWidth": 3.2,
      }
    }
  />
  <div
    style={
      Object {
        "borderColor": "#ffffff",
        "borderWidth": 3.2,
      }
    }
  />
</div>
`;
